src/main/java/net/oauth/jsontoken/JsonToken.java

```diff
@@ -254,10 +256,17 @@ public SignatureAlgorithm getSignatureAlgorithm() {
 
     JsonElement algorithmName = header.get(ALGORITHM_HEADER);
     if (algorithmName == null) {
-      throw new IllegalStateException("JWT header is missing the required '" +
-          ALGORITHM_HEADER + "' parameter");
+      throw new IllegalStateException(
+          "JWT header is missing the required '" + ALGORITHM_HEADER + "' parameter",
+          new InvalidJsonTokenException(ErrorCode.BAD_HEADER));
+    }
+
+    try {
+      return SignatureAlgorithm.getFromJsonName(algorithmName.getAsString());
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(
+          new InvalidJsonTokenException(ErrorCode.UNSUPPORTED_ALGORITHM));
```

Comment:
Does it make sense to make the original exception e as a cause in InvalidJsonTokenException? Or at least keep the error message in the new exception.